hudi-common/src/main/java/org/apache/hudi/internal/schema/convert/AvroInternalSchemaConverter.java

```diff
@@ -73,6 +77,75 @@ public static Schema convert(InternalSchema internalSchema, String name) {
     return buildAvroSchemaFromInternalSchema(internalSchema, name);
   }
 
+  public static InternalSchema pruneAvroSchemaToInternalSchema(Schema schema, InternalSchema originSchema) {
```

Comment:
To clarify, is this only used for internal schema?  Does schema evolution incur record conversion between Row and Avro records (which should be avoided as much as possible)?

Reply:
This is just an operation on the schema itself.
to answer your other question, no, schema evolution does not incur record conversion